unified-installer/astra-unified-installer.sh

```diff
@@ -2590,16 +2714,28 @@ step_generate_torc_patch() {
     fi
 
     # Update image pull secrets if needed
-    if [ -n "$IMAGE_PULL_SECRET" ]; then
-        if echo "$_EXISTING_TORC_PULL_SECRETS" | grep -q "^${IMAGE_PULL_SECRET}$" &> /dev/null; then
-            logdebug "image pull secret '$IMAGE_PULL_SECRET' already present in torc"
-        else
-            if [ -z "$_EXISTING_TRIDENT_OPERATOR_PULL_SECRETS" ]; then
-                torc_patch_list+='{"op":"replace","path":"/spec/imagePullSecrets","value":['"$IMAGE_PULL_SECRET"']},'
+    local -a pull_secrets_to_add=()
+    local -a pull_secrets_to_check=("$IMAGE_PULL_SECRET")
+    find_astra_registry_in_trident_images &> /dev/null && pull_secrets_to_check+=("$__ASTRA_REGISTRY_IMAGE_PULL_SECRET")
+
+    for secret in "${pull_secrets_to_check[@]}" ; do
+        if [ -n "$secret" ]; then
+            if echo "$_EXISTING_TORC_PULL_SECRETS" | grep -q "^${secret}$" &> /dev/null; then
+                logdebug "image pull secret '$secret' already present in torc"
             else
-                torc_patch_list+='{"op":"add","path":"/spec/imagePullSecrets/-","value":"'"$IMAGE_PULL_SECRET"'"},'
+                pull_secrets_to_add+=("$secret")
             fi
         fi
+    done
+    if ! is_empty "${pull_secrets_to_add[@]}"; then
+        if [ -z "$_EXISTING_TRIDENT_OPERATOR_PULL_SECRETS" ]; then
+            local -r secrets_list="$(join_str "${pull_secrets_to_add[@]}" ",")"
+            torc_patch_list+='{"op":"replace","path":"/spec/imagePullSecrets","value":['"$secrets_list"']},'
+        else
+            for secret in "${pull_secrets_to_add[@]}" ; do
+                torc_patch_list+='{"op":"add","path":"/spec/imagePullSecrets/-","value":"'"$secret"'"},'
+            done
+        fi
```

Comment:
The main changes around these parts is determining which (if any) secret out of `IMAGE_PULL_SECRET` and `astra-regcred` we want to include in the list of pull secrets. This involves doing the following for both:
- Check if the secret already exists in the TORC's list of secrets
- If not, add the secret to the list